ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -373,6 +373,11 @@ private[ember] object Parser {
     def parser[F[_]: Concurrent](maxHeaderSize: Int)(
         buffer: Array[Byte],
         read: Read[F],
+    ): F[(Response[F], Drain[F])] = parser[F](maxHeaderSize, discardBody = false)(buffer, read)
+
+    def parser[F[_]: Concurrent](maxHeaderSize: Int, discardBody: Boolean)(
```

Comment:
Is `discardBody` really the right name? AFAICT there is no discarding happening, it seems more similar to `expectNoBody`. Perhaps we can have `requestExpectsNoBody` and `statusExpectsNoBody`.

Reply:
Sorry, I did some more reading.

> Warning: A response to a HEAD method should not have a body. If it has one anyway, that body must be ignored: any [representation headers](https://developer.mozilla.org/en-US/docs/Glossary/Representation_header) that might describe the erroneous body are instead assumed to describe the response which a similar GET request would have received.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

So it does seem like the body must be discarded if it's present. So now I'm not sure if it's appropriate to share a codepath with `statusExpectsNoBody`.